learning_journal_basic1/views/default.py

```diff
@@ -0,0 +1,98 @@
+from pyramid.response import Response
+from pyramid.view import view_config
+from sqlalchemy.exc import DBAPIError
+
+from pyramid.httpexceptions import HTTPFound
+from datetime import datetime
+
+from ..models import MyModel
+
+TIME_FORMAT = '%B %d, %Y'
+
+
+@view_config(route_name='home', renderer='../templates/list.jinja2')
+def home_view(request):
+    """Display a list of all entries in the database on Home Page."""
+    try:
+        query = request.dbsession.query(MyModel)
+        entries = query.all()
+    except DBAPIError:
+        return Response(db_err_msg, content_type='text/plain', status=500)
+    return {"entries": entries}
+
+
+@view_config(route_name='detail_view', renderer='../templates/detail.jinja2')
+def detail_view(request):
+    """Display details of a particular entry based on id."""
+    query = request.dbsession.query(MyModel)
+    entry = query.filter(MyModel.id ==
+                                    int(request.matchdict['id'])).first()
+    return {'entry': entry}
+
+
+@view_config(route_name='entry_view', renderer='../templates/entry.jinja2')
+def entry_view(request):
+    """Display an empty form on GET.
+    Create a new entry, a new model, and return to the Home on POST.
+    Display an error message if inputs title/body are left empty.
+    """
+
+    if request.method == 'GET':
+        return {}
```

Comment:
This looks a bit odd to an experienced web app developer.  In general the pattern for a view that handles `GET` and `POST` requests looks like this:

```python
def pseudoview(request):
    do_shared_things_here() # actions that happen both on GET and POST
    if request.method == 'POST':
        do_post_things_here() # actions that happen only on POST
        return HTTPFound(new_location) # if you want visitors to go elsewhere after POST processing is done
    do_get_things_here() # actions that should only happen on GET (assumes that you redirected after POST)
    return context_for_template
```